### PR TITLE
Install python3.11-venv and gcc-9 in cpubuilder dockerfile.

### PR DESCRIPTION
**Pull request by ScottTodd**
This should help get the byo_llvm and gcc builds working: https://github.com/iree-org/iree/pull/18356 .

Also grouped the installed packages a bit.

Tested with

```bash
# Host
sudo docker buildx build --file dockerfiles/cpubuilder_ubuntu_jammy_x86_64.dockerfile . --tag scotttodd/cpubuilder:latest
sudo docker run --rm --mount type=bind,source="/home/nod/dev/projects/iree",target=/iree -it --entrypoint bash scotttodd/cpubuilder:latest

# Docker
cd iree
./build_tools/cmake/build_and_test_byo_llvm.sh
```

### CODE REVIEW

File: dockerfiles/cpubuilder_ubuntu_jammy_x86_64.Dockerfile [hunk 1]
```diff
@@ -4,17 +4,25 @@ FROM ubuntu:jammy
...
-    apt clean && rm -rf /var/lib/apt/lists/*
+        libcapstone-dev libtbb-dev libzstd-dev
+# Python
+RUN apt install -y python3.11-dev python3.11-venv python3-pip && \
+    update-alternatives --install /usr/local/bin/python python /usr/bin/python3.11 3 && \
+    update-alternatives --install /usr/local/bin/python3 python3 /usr/bin/python3.11 3
+# Toolchains and build deps
+RUN apt install -y \
+        wget git unzip curl gnupg2 lsb-release \
+        ccache ninja-build clang-14 lld-14 gcc-9 g++-9 \
```
Comment (ScottTodd) [inline dockerfiles/cpubuilder_ubuntu_jammy_x86_64.Dockerfile:18]: gcc-11 was preinstalled on `ubuntu:jammy`. We could bump our min/tested version but this matches what IREE currently tests in `.github/workflows/ci_linux_x64_gcc.yml`.

Could also install a matrix of versions like what GitHub provides: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#language-and-runtime

**Review by marbre — APPROVED** [review]
Just two minor nits. It doesn't hurt to have the packages in the list twice, but I would prefer to have not :)

File: dockerfiles/cpubuilder_ubuntu_jammy_ghr_x86_64.Dockerfile [hunk 2]
```diff
@@ -4,17 +4,25 @@ FROM docker.io/myoung34/github-runner:ubuntu-jammy
...
-        libcapstone-dev libtbb-dev libzstd-dev && \
-    apt clean && rm -rf /var/lib/apt/lists/*
+        libcapstone-dev libtbb-dev libzstd-dev
+# Python
+RUN apt install -y python3.11-dev python3.11-venv python3-pip && \
+    update-alternatives --install /usr/local/bin/python python /usr/bin/python3.11 3 && \
+    update-alternatives --install /usr/local/bin/python3 python3 /usr/bin/python3.11 3
+# Toolchains and build deps
+RUN apt install -y \
+        wget git unzip curl gnupg2 lsb-release \
```
Comment (marbre) [inline dockerfiles/cpubuilder_ubuntu_jammy_ghr_x86_64.Dockerfile:17]:
```suggestion
```
Those are already installed as part of the basic apt packages. Either remove here (preferred) or above.

File: dockerfiles/cpubuilder_ubuntu_jammy_x86_64.Dockerfile [hunk 3]
```diff
@@ -4,17 +4,25 @@ FROM ubuntu:jammy
...
-        libcapstone-dev libtbb-dev libzstd-dev && \
-    apt clean && rm -rf /var/lib/apt/lists/*
+        libcapstone-dev libtbb-dev libzstd-dev
+# Python
+RUN apt install -y python3.11-dev python3.11-venv python3-pip && \
+    update-alternatives --install /usr/local/bin/python python /usr/bin/python3.11 3 && \
+    update-alternatives --install /usr/local/bin/python3 python3 /usr/bin/python3.11 3
+# Toolchains and build deps
+RUN apt install -y \
+        wget git unzip curl gnupg2 lsb-release \
```
Comment (marbre) [inline dockerfiles/cpubuilder_ubuntu_jammy_x86_64.Dockerfile:17]:
```suggestion
```
Same as above.

File: dockerfiles/cpubuilder_ubuntu_jammy_ghr_x86_64.Dockerfile (hunk 2 above)
Comment (ScottTodd) [reply dockerfiles/cpubuilder_ubuntu_jammy_ghr_x86_64.Dockerfile]: Oops, good catch.

**Pull request merged**